CMakeLists.txt

```diff
@@ -22,6 +22,10 @@ option(BUILD_TESTS "Compile the tests" OFF)
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 option(WITH_TENSOR_PARALLEL "Compile with NCCL and MPI backend" OFF)
 option(WITH_FLASH_ATTN "Compile with Flash Attention 2" OFF)
+option(GOOGLE_ADDRESS_SANITIZER "ASAN" OFF)
+
```

Comment:
Two suggestions:
- ENABLE_ADDRESS_SANITIZER may be a better name than GOOGLE_ADDRESS_SANITIZER because it is not Google specific (you also changed the name of the PR).
- Consider a better description for the switch like "Enable AddressSanitizer for memory error detection"